src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs

```diff
@@ -12,11 +13,56 @@ namespace Microsoft.IdentityModel.Tokens
     /// </summary>
     public class SecurityTokenDescriptor
     {
+        // TODO: At next major version (8.0), remove Audience and logic for combining with Audiences.
+        private HashSet<string> _audiences;
+
         /// <summary>
-        /// Gets or sets the value of the 'audience' claim.
+        /// Gets or sets the value of the 'audience' claim. Will be deprecated in favor of <see cref="Audiences"/> in the next
+        /// major version (8.x).
         /// </summary>
         public string Audience { get; set; }
 
+        /// <summary>
+        /// Gets or sets one or more audiences to include in the token's 'Aud' claim. Automatically removes duplicates and empty,
+        /// null, or whitespace-only strings. Does not use a threadsafe collection.
+        /// </summary>
+        public IEnumerable<string> Audiences {
```

Comment:
This should be an IList<string> and have a lazy creation pattern here: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/a1530a0cebd47793979ad6e316622013acb3d9ef/src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs#L280

Reply:
When I spoke with @jmprieur and others we aligned on IEnumerable in part to match other APIs and in part to match [the guidance here](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections#collection-parameters). What is the right forum to make public API decisions?

Reply:
Changing to IList